protokolle/2020-04-29_htw-pv3_postgresql.md

```diff
@@ -0,0 +1,265 @@
+
+  4. * Disconnected vom Server
+	      
+  5. * PostgresSQL 12: Properties ändern
+
+     * user == sonnja
+
+  6. * reconnecten mit Passwort : sonnja
+
+
+* FRAGE: Wie wurde die Datenbank "sonnja_db" hinzugefügt? Steht das in dem Skript 2?
+
+* -> Ja (Aber Prinzipiell Rechtsklick auf die Datenbank und neue Datenbank erzeugen)
+
+
+
+* User Management mit Skript (von LH) erstellen:
+
+  * user Rolle erstellen
+
+  * admin Rolle erstellen
+
+
+# 17:00 Daten-Session
+
+    Welche Daten haben wir?
+
+    METADATEN erstellen
+
+    Datenbankimport versuchen
+
+
+ [LINK](https://next.rl-institut.de/s/LJXYzBDyC5aDJtz) zu Rohdaten
+
+
+# Aufgaben
+
+
+   - [] Metadaten zu den CSV-Dateien "HTW-Wetter" und "Sonja-Messdaten" generieren [ALL]
+   
+   - [] die richtigen Datenbankbefehle für die Fehler -> Lücken finden und Lücken schließen [ALL]
+   
+   ## 1. SQL-Script
+   
+   -- Groups
+DROP ROLE IF EXISTS role_read; 
+CREATE ROLE role_read WITH NOLOGIN NOSUPERUSER NOCREATEDB NOCREATEROLE INHERIT NOREPLICATION; -- Read only
+
+DROP ROLE IF EXISTS role_user; 
+CREATE ROLE role_user WITH NOLOGIN NOSUPERUSER NOCREATEDB NOCREATEROLE INHERIT NOREPLICATION; 
+GRANT role_read TO role_user; -- Insert, Update, Delete and Read
+
+DROP ROLE IF EXISTS role_admin; 
+CREATE ROLE role_admin WITH NOLOGIN SUPERUSER CREATEDB CREATEROLE REPLICATION; 
+GRANT role_user TO role_admin; -- Admin, can create user and databases
+
+-- User
+DROP USER IF EXISTS user_read; 
+CREATE USER user_read LOGIN PASSWORD 'read' NOSUPERUSER INHERIT NOCREATEDB CREATEROLE REPLICATION;
+GRANT role_read TO user_read; -- 2017-04-17 LH
+
+DROP USER IF EXISTS user_write; 
+CREATE USER user_write LOGIN PASSWORD 'write' NOSUPERUSER INHERIT NOCREATEDB CREATEROLE REPLICATION; 
+GRANT role_user TO user_write; -- 2017-04-17 LH
+
+DROP USER IF EXISTS user_admin; 
+CREATE USER user_admin LOGIN PASSWORD 'admin' SUPERUSER INHERIT CREATEDB CREATEROLE REPLICATION; 
+GRANT role_admin TO user_admin; -- 2017-04-17 LH
+
+DROP USER IF EXISTS sonnja; 
+CREATE USER sonnja LOGIN PASSWORD 'sonnja' SUPERUSER INHERIT CREATEDB CREATEROLE REPLICATION; 
+GRANT role_admin TO sonnja; -- 2018-04-25 LH
+
+
+   
+
+
+## Fragen zu Git:
+
+* -commit auf falschen branch a
+
+* -wurde außerdem auf anderem branch b überholt
+
+* -branch a ist aber noch nicht abgeschlossen
+
+* -[Link](https://github.com/htw-pv3/literature/issues/11)
+
+* -guter Lösungsweg?
+
+        
+* Sobald bei git nicht trivale Unterschiede auftauchen (z.B. eine Bearbeitung in der gleichen Zeit), tritt ein sognannter "Merge Conflict" auf.
+* Das sieht zunächst schlimmer aus als es wirklich ist.
+* Das macht kurz Angst, kann aber relativ einfach behoben werden.
+* Ich schau es mir mal in Ruhe an und zeige eine einfache Lösung.
+* Ich verwende das Programm Meld (http://meldmerge.org/) zum vergleichen von Dokumenten (Dif-Tool). Damit ist ein "manuelles Beheben" sehr einfach.
+
```

Comment:
- rework structure. Not every line is a bullet point
- remove - after *
- find a better discription for the link like "Link to Issue"

Reply:
Also done :-)